packages/babel-helper-split-export-declaration/src/index.js

```diff
@@ -0,0 +1,65 @@
+import * as t from "@babel/types";
+
+const splitExportDeclaration = exportDeclaration => {
+  if (!exportDeclaration.isExportDeclaration()) {
+    throw new Error("Only export declarations can be splitted.");
+  }
+
+  // build specifiers that point back to this export declaration
+  const isDefault = exportDeclaration.isExportDefaultDeclaration();
+  const declaration = exportDeclaration.get("declaration");
+
+  if (isDefault) {
+    if (declaration.isIdentifier()) {
+      throw new Error("It doesn't make sense to split exported identifier.");
+    }
+    const standaloneDeclaration =
+      declaration.isFunctionDeclaration() || declaration.isClassDeclaration();
+
+    let id = declaration.node.id;
+    const liveBinding = !!id;
```

Comment:
```js
export default function() {}
```
is transformed into
```js
function _default() {}
export default _default;
```

The first export statement is hoisted, but is it also the second?

Reply:
could u elaborate? I've thought about hoisting briefly in this context, but came to conclusion that no extra work needs to be done for export statements, they are evaluated in place, right? and as function declaration will get hoisted by javascript this identifier will be available by the time it gets exported

Reply:
After playing a bit with it, I understood that the way you implemented it is correct 👍